### PR TITLE
Cli Usability: Cleanup fn commands and make dir optional

### PR DESCRIPTION
**Pull request by phanimarupaka**
This Pr does the following

1. Make DIR field optional for all fn commands.
2. `kpt fn source` accepts input only from a directory and not stdin
3. `kpt fn eval` accepts stdin iff `-` is specified making it consistent with kubectl and other commands
4. `kpt fn sink` only writes to output Dir and not to stdout.

Example of out of place hydration

```
kpt fn source | kpt fn eval - --image gcr.io/kpt-fn/apply-setters -- environment=dev | kpt fn sink dev/
```

In-place hydration

```
kpt fn eval --image gcr.io/kpt-fn/apply-setters -- environment=dev
```

### CODE REVIEW

File: thirdparty/cmdconfig/commands/cmdeval/cmdeval.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ import (
 func GetEvalFnRunner(name string) *EvalFnRunner {
 	r := &EvalFnRunner{}
 	c := &cobra.Command{
-		Use:     "eval [DIR]",
+		Use:     "eval [DIR | -<STDIN>]",
```
Comment (droot) [inline thirdparty/cmdconfig/commands/cmdeval/cmdeval.go:24]: Do we have to say `STDIN` here ?

**Review by droot — APPROVED** [review]

File: thirdparty/cmdconfig/commands/cmdeval/cmdeval.go (hunk 1 above)
Comment (phanimarupaka) [reply thirdparty/cmdconfig/commands/cmdeval/cmdeval.go:24]: What are you suggesting ?

`eval [DIR]` or
`eval [DIR | -]`

Comment (droot) [reply thirdparty/cmdconfig/commands/cmdeval/cmdeval.go:24]: I am thinking.. just saying `eval [DIR | -]` should be enough..right ?

Comment (phanimarupaka) [reply thirdparty/cmdconfig/commands/cmdeval/cmdeval.go]: Done.

**Comment by frankfarzan** [comment]
@Shell32-Natsu @droot We don't have sufficient e2e test coverage for piping behavior `source | eval | sink` today, which needs to be added soon to make sure there's no regression here.

File: site/reference/pkg/tree/README.md [hunk 2]
```diff
@@ -97,7 +97,7 @@ kubectl get all -o yaml | kpt pkg tree \
 ### Synopsis
 <!--mdtogo:Long-->
 ```
-kpt pkg tree [DIR | -<STDIN>] [flags]
+kpt pkg tree [DIR | -] [flags]
```
Comment (frankfarzan) [inline site/reference/pkg/tree/README.md:100]: Where is the Synopsis do we explain what `-` means? This applies to all commands not just this.

Comment (phanimarupaka) [reply site/reference/pkg/tree/README.md:100]: I will do the reference docs clean-up in coming PRs.

**Comment by phanimarupaka** [comment]
This PR is waiting on https://github.com/GoogleContainerTools/kpt/pull/1711 to be merged.

**Review by frankfarzan — APPROVED** [review]

**Pull request merged**